spec/src/main/asciidoc/configuration.asciidoc

```diff
@@ -238,14 +251,23 @@ MicroProfile JWT implementations are required to throw a `DeploymentException` i
 [[verification-publickey-location]]
 #### `mp.jwt.verify.publickey.location`
 
-The `mp.jwt.verify.publickey.location` config property allows for an external or internal location
+The `mp.jwt.verify.publickey.location` configuration property allows for an external or internal location
 of Public Verification Key to be specified.  The value may be a relative path or a URL.
 
 MicroProfile JWT implementations are required to check the path at startup or deploy time.
 Reloading the Public Key from the location at runtime as well as the frequency of any such
 reloading is beyond the scope of this specification and any such feature should be
 considered vendor-specific.
 
+#### `mp.jwt.verify.publickey.algorithm`
+
+The `mp.jwt.verify.publickey.algorithm` configuration property allows for specifying which Public Key Signature Algorithm
+is supported by the MP JWT endpoint. This property can be be set to either `RS256` or `ES256`. Default value is `RS256`.
+
+`mp.jwt.verify.publickey.algorithm` will provide an additional hint how to read the Public Key (for example, EC Public Key in the PKCS#8 PEM format will only have a standard `-----BEGIN PUBLIC KEY-----` header and footer). It will also help avoiding the security risks associated with the token `alg`(algorithm) header dictating to the MP JWT endpoint which signature verification algorithm should be chosen.
```

Comment:
But this `alg` claim check is a breaking change, so ES256 support only in 2.0?

Reply:
There is no any requirement here for the implementations to check `alg`manually - it is done by the underlying JOSE library. But what it says, now that 2 algorithms are supported, that using an `alg` check to determine which key to load is a security risk and the algorithm should be whitelisted before the verification starts - the point is the endpoint has a static upfront knowledge about the supported algorithm.
Also this property in itself does not change anything with respect to the way PEM keys have to be processed. Even now if someone puts an EC key into PEM then the implementations trying to get `RSAPublicKey` has to fail. Few things have to be tightened about the way the invalid keys are reported (recall we are having another discussion in relation to URL based keys, etc), but I'd rather prefer not to overload this PR. 
As such it is also a hint. If we have a PEM key with only `-----BEGIN PUBLIC KEY-----` then it can be either RSA or EC key. The implementations may want to use this property to optimize (ex, immediately read it as EC key as opposed to 1) read it as RSA key first and if it fails 2) as EC key). RSA PEM keys may have `RSA` qualifiers. So it is mainly a hint for reading the PEM keys. 

Yes, if we have a JWK set then this property provides for an additional `kty` check, apart from the matching `kid` requirement.

Let me try to simplify and rework the text a bit